app/jobs/indexing_job.rb

```diff
@@ -0,0 +1,18 @@
+##
+# Base IndexingJob class
+class IndexingJob < ActiveJob::Base
+  queue_as :indexing
+
+  ##
+  # Perform indexing job
+  # @param [Array] pid_list
+  # @param [Boolean] should_commit
+  # @param [Boolean] should_profile
+  def perform(pid_list, should_commit = false, should_profile = false)
```

Comment:
The `should_profile` could be removed and a `Settings` variable used instead to indicate profiling. That way queued jobs aren't pinned to profiling... or the next line could be something like `if should_profile && Settings.ENABLE_PROFILING`